samples/highcharts/demo/area-race-chart/demo.js

```diff
@@ -0,0 +1,321 @@
+            ]
+        }
+    ],
+
+    responsive: {
+        rules: [
+            {
+                condition: {
+                    maxWidth: 500
+                },
+                chartOptions: {
+                    subtitle: {
+                        y: -150,
+                        x: -20
+                    },
+                    yAxis: {
+                        title: {
+                            align: 'high',
+                            rotation: 0,
+                            offset: -80,
+                            x: 0,
+                            y: -20
+                        }
+                    }
+                }
+            }
+        ]
+    }
+});
+
+function pause(button) {
+    button.title = 'play';
+    button.className = 'fa fa-play';
+    clearTimeout(chart.sequenceTimer);
+    chart.sequenceTimer = undefined;
+}
+
+function update() {
+    chart.update(
+        {
+            subtitle: {
+                text: getSubtitle()
+            }
+        },
+        false,
+        false,
+        false
+    );
+
+    const series = chart.series,
+        labels = chart.annotations[0].labels,
+        yearIndex = input.value - startYear,
+        dataLength = series[0].data.length;
+
+    // If slider moved back in time
+    if (yearIndex < dataLength - 1) {
+        for (let i = 0; i < series.length; i++) {
+            const seriesData = series[i].data.slice(0, yearIndex);
+            series[i].setData(seriesData, false);
+        }
+    }
+
+    // If slider moved forward in time
+    if (yearIndex > dataLength - 1) {
+        const remainingYears = yearIndex - dataLength;
+        for (let i = 0; i < series.length; i++) {
+            for (let j = input.value - remainingYears; j < input.value; j++) {
+                series[i].addPoint([formatRevenue[i][j] || 0], false);
+            }
+        }
+    }
+
+    // Add current year
+    for (let i = 0; i < series.length; i++) {
+        const newY = formatRevenue[i][input.value] || 0;
+        series[i].addPoint([newY], false);
+    }
+
+    labels.forEach(label => {
+        label
+            .graphic
+            .css({
+                opacity: input.value >= label.options.point.x | 0
+            });
+    });
+
+    chart.redraw();
+
+    input.value = parseInt(input.value, 10) + 1;
+
+    if (input.value > endYear) {
+        // Auto-pause
+        pause(btn);
+    }
+}
+
+function play(button) {
+    // Reset slider at the end
+    if (input.value > endYear) {
+        input.value = startYear;
+    }
+    button.title = 'pause';
+    button.className = 'fa fa-pause';
+    chart.sequenceTimer = setInterval(function () {
+        update();
+    }, 700);
+}
+
+btn.addEventListener('click', function () {
+    if (chart.sequenceTimer) {
+        pause(this);
+    } else {
+        play(this);
+    }
+});
+
+play(btn);
+
+// Trigger the update on the range bar click.
+input.addEventListener('input', function () {
+    update();
+});
```

Comment:
Could be simplified to `input.addEventListener('input', update);` as far as I can see